dashboard/app/models/programming_environment.rb

```diff
@@ -9,5 +9,34 @@
 #  updated_at :datetime         not null
 #
 class ProgrammingEnvironment < ApplicationRecord
+  include SerializedProperties
+
   has_many :programming_expressions
+
+  serialized_attrs %w(
+    editor_type
+  )
```

Comment:
is this an enum, or a free-form string? can you add a class-level comment indicating the valid values for editor_type? If you want you can follow the syntax (for example) at the top of activity_section.rb